### PR TITLE
Feature/198 - add Exercises

### PR DESCRIPTION
**Pull request by SaXaPhonist**
- add exercise to materials of courses
- create migration
- add mongoose validation 
- refactor component to Exercise

### CODE REVIEW

File: frontend/src/pages/LearningCourse/Exercise/Exercise.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  CourseInfoContent,
+  CourseInfoTitle,
+  CourseInfoText,
+  CodeBlock,
+} from './styled';
+
+interface IExerciseProps {
+  courseExersice: IExercise;
+  isCourseInfoOpen: boolean;
+}
```
Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/Exercise/Exercise.tsx:16]: please move to types folder with index.ts inside it

File: frontend/src/pages/LearningCourse/Exercise/styled.ts [hunk 2]
```diff
@@ -64,9 +67,21 @@ export const CourseInfoTitle = styled('span')({
   },
 });
 
+export const CodeBlock = styled('code')({
+  fontSize: '14px',
+  display: 'block',
+  backgroundColor: '#eee',
```
Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/Exercise/styled.ts:73]: uppercase

Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/Exercise/styled.ts:73]: uppercase

File: frontend/src/pages/LearningCourse/LearningCourse.tsx [hunk 3]
```diff
@@ -5,14 +5,14 @@ import { AuthorizedLayout } from 'components/Layout';
...
 
 import ActionButton from './ActionButton';
-import CourseInfo from './CourseInfo';
 import CourseInfoToggle from './CourseInfoToggle';
 import Material from './Material';
 import StageController from './StageController';
 import { LearningPageContainer, BackButton, LearningWrapper } from './styled';
+import Exercise from './Exercise';
 
 interface IProps {
```
Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/LearningCourse.tsx:17]: please move to local types folder

Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/LearningCourse.tsx:17]: please move to types folder

File: frontend/src/pages/LearningCourse/LearningCourseContainer.tsx [hunk 4]
```diff
@@ -70,7 +70,7 @@ const LearningCourseContainer: React.FC = () => {
 
   const { materials: courseMaterials } = courseMaterialsResponse;
 
-  const courseMaterial = courseMaterials[stage - 1]?.content[CONTENT_ELEMENT];
+  const courseContent = courseMaterials[stage - 1]?.content[CONTENT_ELEMENT];
```
Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/LearningCourseContainer.tsx:73]: magic number

Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/LearningCourseContainer.tsx:73]: magic number

File: frontend/src/pages/LearningCourse/LearningCourseContainer.tsx [hunk 5]
```diff
@@ -91,11 +91,12 @@ const LearningCourseContainer: React.FC = () => {
   };
   return (
     <LearningCourse
-      key={courseMaterialsResponse.materials[stage - 1]._id}
+      key={courseMaterialsResponse.materials[stage - 1].stage}
```
Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/LearningCourseContainer.tsx:94]: magic number

Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/LearningCourseContainer.tsx:94]: magic number

File: frontend/src/pages/LearningCourse/LearningCourseContainer.tsx [hunk 6]
```diff
@@ -91,11 +91,12 @@ const LearningCourseContainer: React.FC = () => {
...
       stage={stage}
       maxStage={maxStage}
       courseInfo={courseInfo}
-      courseMaterial={courseMaterial}
+      courseMaterial={courseMaterials[stage - 1]}
```
Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/LearningCourseContainer.tsx:98]: magic number

Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/LearningCourseContainer.tsx:98]: magic number

File: frontend/src/pages/LearningCourse/Material/MaterialContainer.tsx [hunk 7]
```diff
@@ -2,12 +2,12 @@ import { FC } from 'react';
 
 import { getPreviewId } from 'utils/helpers/videoPlayer/getPreviewId';
 import extractMaterial from 'utils/helpers/materials/extractMaterial';
-import { IContentElement } from 'types/course';
+import { TContentElement } from 'types/course';
 
 import Material from './Material';
 
 interface IProps {
```
Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/Material/MaterialContainer.tsx:9]: local types folder

Comment (irina-sinitsyna) [inline frontend/src/pages/LearningCourse/Material/MaterialContainer.tsx:9]: local types folder

File: frontend/src/types/course.ts [hunk 8]
```diff
@@ -1,19 +1,26 @@
 import { COURSE_LABELS } from 'constants/statuses';
 import { CourseStatus } from 'enums/course';
 import { EmployeeRank } from 'enums/employee';
+import { TMaterialVariants } from './materials';
```
Comment (irina-sinitsyna) [inline frontend/src/types/course.ts:4]: imports order

Comment (irina-sinitsyna) [inline frontend/src/types/course.ts:4]: imports order

File: frontend/src/types/course.ts [hunk 9]
```diff
@@ -1,19 +1,26 @@
...
-    content: IContentElement[];
-    isCompleted: boolean;
-  }[];
-  technology: Array<string>;
-  _id?: string;
+export type TContentElement = {
+  type: TMaterialVariants;
+  material: string;
 };
+export interface IExercise {
```
Comment (irina-sinitsyna) [inline frontend/src/types/course.ts:12]: add an empty line here please

Comment (irina-sinitsyna) [inline frontend/src/types/course.ts:12]: add an empty line here please

File: frontend/src/pages/LearningCourse/Exercise/Exercise.tsx (hunk 1 above)
Comment (SaXaPhonist) [reply frontend/src/pages/LearningCourse/Exercise/Exercise.tsx:16]: i think that it's odd

**Review by dapodev — APPROVED** [review]

File: backend/migrations-backdev/20220125143932-backdev_col.js [hunk 10]
```diff
@@ -1446,18 +1447,16 @@ module.exports = {
       },
     );
     await db.createCollection('clientCourses');
-    await db.createCollection('accessTokenBlacklist');
```
Comment (k-krasilnikova) [inline backend/migrations-backdev/20220125143932-backdev_col.js:1449]: Why removing this collection?

Comment (SaXaPhonist) [reply backend/migrations-backdev/20220125143932-backdev_col.js:1449]: idk, i took some data from @zhuk-vezunchikz  migration

**Review by dapodev — APPROVED** [review]

**Review by dapodev — APPROVED** [review]

**Pull request merged**